### ISSUE TITLE
Why we need 

### ISSUE THREAD

**Issue by liming-ai** [Chumsy0725/logit-adj-pytorch#3]
Hi @sadeepj , Thanks for your amazing work, I noticed that [there](https://github.com/Chumsy0725/logit-adj-pytorch/blob/7ef352fc9d2bac4f9c31cea0c2091f6c089bfaff/main.py#L120) is another weight decay term:
```
loss = loss + args.weight_decay * loss_r
```

Why do we need this line? The weight decay has been set in [optimizer init](https://github.com/Chumsy0725/logit-adj-pytorch/blob/7ef352fc9d2bac4f9c31cea0c2091f6c089bfaff/main.py#L59)

Besides, I did not found any weight decay in the official TensorFlow code. Could you please explain it?
